_posts/2021-10-01-gs_00004_basic_shell_script.md

```diff
@@ -0,0 +1,318 @@
+
+    firefox -new-tab -url bitsandvolts.org \
+            -new-tab -url https://github.com/bitsandvolts/getting_started &
+
+<div class='note-normal' markdown="1">
+_Note: For better readability, we split the command over two lines using '\\' You can also write it into one line without '\\'_.
+</div>
+
+Save the script and run it again:
+
+    $ ./open_firefox.sh
+
+A tab to our blog and a tab to the git repository should be opened.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+Can you imagine another program you want to run for your daily work?
+
+</div>
+
+# A second example
+
+Congratulations! You just ran your first script! Good job! Now we want to give
+you a second example and explain two more interesting features of bash scripts.
+Imagine you are working in a company with multiple projects. We often experience
+that every employee organizes the projects he or she is responsible for in
+his/her own way. This, however, has a huge drawback. If there is a new
+colleague, he/she has to find his/her own way. Moreover, if you take over the
+project from a colleague you have to spend time to understand his/her project
+organization. A better way would be to have a project template that everyone
+uses. Let's create a very basic one!
+
+First, you usually would create a folder with the project name. Then, inside
+this folder, you might have a folder for the project sources and one for the
+documentation. Additionally, there might be a **README** and a **LICENSE** file.
+In the REAMDE you usually put basic information. For example, how you can
+install the software or how your project is organized. The LICENSE file is
+important if you want to publish your project. It tells others how they are
+allowed to use it.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+Check our [GitHub repository](https://github.com/bitsandvolts/getting_started)
+and find an example for a README and a LICENSE file.
+
+</div>
+
+From the previous posts and the first bash script above you should already know
+how to do it. Use nano to open a file 'project_template.sh' and enter the
+following:
+
+    #!/bin/bash
+
+    # Print something to the terminal
+    echo "Greetings from Bits&Volts (bitsandvolts.org)"
+
+    # Create the project folder
+    mkdir my_project
+
+    # move into the project folder
+    cd my_project
+
+    # create other folders and files
+    mkdir sources
+    mkdir documentation
+    touch README.md
+    touch LICENSE
+
+<div class='note-normal' markdown="1">
+_Note: '.md' is a Markdown extension, which is typical for a README. We will talk about it in a future post._
+</div>
+
+Save the file, add the correct privileges and run it:
+
+    $ chmod +x project_template.sh
+    $ ./project_template.sh
+
+If you list your directory contents, you should see the created project:
+
+    $ ls
+
+So far so good! However, not all projects are called 'my_project' and it would
```

Comment:
```suggestion
So far, so good! However, not all projects are called 'my_project' and it would
```
Actually, I checked online, and there people also using "so far so good" without comma. So decide by yourself what is better (I also made a "correction" of this somewhere above)